### ISSUE TITLE
Samtools View with * Region

### ISSUE THREAD

**Issue by amanpatel101** [samtools/samtools#867]
#### Are you using the latest version of [samtools]
Yes
 #### Please describe your environment.
- OS : Darwin 17.5.0
- machine architecture: x86_64
- compiler:
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
GB-C02T41T4HP49:~ ampatel$ clang --version
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

#### Please specify the steps taken to generate the issue, the command you are running and the relevant output.
I have some bam files from which I am trying to extract the unmapped reads. samtools idxstats shows several unmapped reads in the '\*' region, which are the ones I seek. The documentation states I can simply run the command samtools view -b [filename] '\*' -o [out.file] to extract these reads. However, when I attempt to convert the output files to fastq files with samtools bam2fq, I am notified that 0 reads were processed. Is it possible that '\*' is not a valid region to specify, even though the documentation says otherwise? Thanks!

**Comment by jkbonfield** [comment]
I just checked this with Samtools18 and it does work.  I tried `samtools view -c` to count the entries and checked these matched the figure reported from idxstat.

Are you quoting it correctly? Without quotes the shell will expand * to mean all the filenames in the current directory.

Also, is this a bam2fq usage issue rather than a samtools view one?  Did you check whether the out.file actually contains any reads?  I would imagine it does, and maybe this is something to do with filtering in bam2fq instead.
